### DISCUSSION TITLE
PostgREST in WebAssembly ?

### DISCUSSION THREAD

**Discussion by nileshtrivedi** [PostgREST/postgrest#2353] [Q&A] [answered]
Has anyone tried porting PostgREST to WASM binary format? This opens the possibility of deploying PostgREST on edge cloud infra like Deno.com - which can be a lot more convenient and cost-effective.

I'm not familiar with Haskell ecosystem. I couldn't find any discussions about WASM in [Stack](https://github.com/commercialhaskell/stack/issues) repo.

**Comment by steve-chavez** [comment]
There's https://github.com/tweag/asterius, they list pandoc as one of the compiled apps, so it might be possible.

**Reply by steve-chavez** [reply]
We depend on `libpq`(C dependency), so that might be a blocker.

There was a discussion about using a pure Haskell library instead of libpq https://github.com/PostgREST/postgrest/issues/2261#issuecomment-1171073341

**Comment by wolfgangwalther** [comment, marked as answer]
This seems to be more of an idea for a feature request than a discussion, so I converted it to #2452.
